librz/il/definitions/bitvector.c

```diff
@@ -0,0 +1,1078 @@
+// SPDX-FileCopyrightText: 2021 heersin <teablearcher@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_il/definitions/bitvector.h>
+
+/**
+ * New a `length`-bits bitvector
+ * \param length int, the length of bitvector
+ * \return bv RzILBitVector, pointer to the new bitvector instance
+ */
+RZ_API RzILBitVector rz_il_bv_new(int length) {
+	RzILBitVector ret = (RzILBitVector)malloc(sizeof(struct bitvector_t));
+	if (ret == NULL) {
+		return NULL;
+	}
+
+	// how much ut8 do we need to represent `length` bits ?
+	int real_elem_cnt = NELEM(length, sizeof(ut8));
```

Comment:
I think this should be moved to `rz_il_bv_init()`.

Reply:
or change the name to rz_il_bv_new0() ?

Reply:
I don't think it's required. Uninitiated bitvector doesn't make sense. It's *always* a stream of bits. If you need unknown value the bitvector shouldn't be used.

Reply:
I would prefer to have `init/fini` APIs to be able to use bitvectors on stack or as part of another struct.

Reply:
Ok. Then it indeed makes sense. Good proposition. cc @Heersin

Reply:
```c
RzBitVector *rz_il_bv_new(ut32 length) {
    RzBitVector *bv = RZ_NEW0(RzBitVector);
    if (!bv) {
        return NULL;
    }
    if (!rz_il_bv_init(bv, length)) {
        RZ_FREE(bv);
    }
    return bv;
}
```

Reply:
Same for `fini`: `rz_bv_free()` invokes `rz_bv_fini()`.